handlers/validator.go

```diff
@@ -2057,6 +2058,9 @@ func ValidatorSync(w http.ResponseWriter, r *http.Request) {
 	}
 
 	totalCount = uint64(len(slots))
+	if start >= totalCount {
+		start = totalCount - 1
+	}
 
 	startIndex := start + length - 1
 	if startIndex > uint64(len(slots)-1) {
```

Comment:
**suggestion:** Could be
```suggestion
	if startIndex >= totalCount{
```
and the line below it `startIndex = totalCount - 1`

Reply:
Done.